contracts/StakeManager.sol

```diff
@@ -157,12 +180,32 @@ contract StakeManager is Ownable {
                 revert StakeManager__InvalidLockTime();
             }
         }
+
         _mintBonusMP(account, deltaTime, _amount);
 
+        //mp estimation
+        uint256 mpPerEpoch = _getMPToMint(_amount, EPOCH_SIZE);
+        if(mpPerEpoch < 1){
+            revert StakeManager__StakeIsTooLow();
+        }
+        uint256 thisEpochExpiredMP = _getMPToMint(_amount, block.timestamp - epochs[currentEpoch].startTime);
+        currentEpochExpiredMP += thisEpochExpiredMP; //TODO: SHIFT ESTIMATION TO LAST EPOCH
+        totalMPPerEpoch += mpPerEpoch;
```

Comment:
Here I'm a bit confused..
At this point, what speaks against doing:

```
totalMPPerEpoch = totalMPPerEpoch + mpPerEpoch - thisEpochExpiredMP;
```

?

Then `totalMPPerEpoch` would already have the missed MPs taken into account. And then I think we also don't need `currentEpochExpiredMP`, do we?

Reply:
Initially I was reducing the value of thisEpochExpiredMP in the totalMPPerEpoch by adding that value in the expiredMPPerEpoch[currentEpoch], but that was hurting the estimation because this value should only be reduced in currentEpoch, not in future epochs.